tests/foreman/virtwho/cli/test_nutanix.py

```diff
@@ -173,3 +173,134 @@ def test_positive_hypervisor_id_option(
+        assert virtwho_config['status'] == 'No Report Yet'
+        command = get_configure_command(virtwho_config['id'], default_org.name)
+        hypervisor_name, guest_name = deploy_configure_by_command(
+            command, form_data['hypervisor-type'], debug=True, org=default_org.label
+        )
+        # Check the option "prism_central=true" should be set in etc/virt-who.d/virt-who.conf
+        config_file = get_configure_file(virtwho_config['id'])
+        assert get_configure_option("prism_central", config_file) == 'true'
+        virt_who_instance = target_sat.cli.VirtWhoConfig.info({'id': virtwho_config['id']})[
+            'general-information'
+        ]['status']
+        assert virt_who_instance == 'OK'
+        hosts = [
+            (hypervisor_name, f'product_id={settings.virtwho.sku.vdc_physical} and type=NORMAL'),
+            (guest_name, f'product_id={settings.virtwho.sku.vdc_physical} and type=STACK_DERIVED'),
+        ]
+        for hostname, sku in hosts:
+            host = target_sat.cli.Host.list({'search': hostname})[0]
+            subscriptions = target_sat.cli.Subscription.list(
+                {'organization': default_org.name, 'search': sku}
+            )
+            vdc_id = subscriptions[0]['id']
+            if 'type=STACK_DERIVED' in sku:
+                for item in subscriptions:
+                    if hypervisor_name.lower() in item['type']:
+                        vdc_id = item['id']
+                        break
+            result = target_sat.cli.Host.subscription_attach(
+                {'host-id': host['id'], 'subscription-id': vdc_id}
+            )
+            assert result.strip() == 'Subscription attached to the host successfully.'
+        target_sat.cli.VirtWhoConfig.delete({'name': virtwho_config['name']})
+        assert not target_sat.cli.VirtWhoConfig.exists(search=('name', form_data['name']))
+
+    @pytest.mark.tier2
+    def test_positive_prism_central_deploy_configure_by_script(
+        self, default_org, form_data, target_sat
+    ):
+        """Verify "hammer virt-who-config fetch" on nutanix prism central mode
+
+        :id: fed77035-8f5e-4d46-a2d5-736b72e1c775
+
+        :expectedresults:
+            Config can be created, fetch and deploy with prism central mode
+            The prism_central has been set in /etc/virt-who.d/vir-who.conf file
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        form_data['prism-flavor'] = "central"
+        virtwho_config = target_sat.cli.VirtWhoConfig.create(form_data)['general-information']
+        assert virtwho_config['status'] == 'No Report Yet'
+        script = target_sat.cli.VirtWhoConfig.fetch(
+            {'id': virtwho_config['id']}, output_format='base'
+        )
+        hypervisor_name, guest_name = deploy_configure_by_script(
+            script, form_data['hypervisor-type'], debug=True, org=default_org.label
```

Comment:
Apart from this block of code, everything is the same in the above test before this , Maybe you can some helper functions as Griffin suggested in your UI pr or 
you can also consider parametrizing such tests

```
@pytest.mark.parametrize('deploy_type', ['id', 'script'])
def test_positive_prism_central_deploy_configure_by_id(self, default_org, form_data, target_sat, deploy_type):
    .
    .
    if deploy_type == "id":
        command = get_configure_command(virtwho_config['id'], default_org.name)
        hypervisor_name, guest_name = deploy_configure_by_command(
            command, form_data['hypervisor-type'], debug=True, org=default_org.label
        )
    elif deploy_type == "script":
        script = target_sat.cli.VirtWhoConfig.fetch(
            {'id': virtwho_config['id']}, output_format='base'
        )
        hypervisor_name, guest_name = deploy_configure_by_script(
            script, form_data['hypervisor-type'], debug=True, org=default_org.label
    .
    .
```

Reply:
> Apart from this block of code, everything is the same in the above test before this , Maybe you can add some helper functions as Griffin suggested in your UI pr or you can also consider parametrizing such tests
> 
> ```
> @pytest.mark.parametrize('deploy_type', ['id', 'script'])
> def test_positive_prism_central_deploy_configure_by_id(self, default_org, form_data, target_sat, deploy_type):
>     .
>     .
>     if deploy_type == "id":
>         command = get_configure_command(virtwho_config['id'], default_org.name)
>         hypervisor_name, guest_name = deploy_configure_by_command(
>             command, form_data['hypervisor-type'], debug=True, org=default_org.label
>         )
>     elif deploy_type == "script":
>         script = target_sat.cli.VirtWhoConfig.fetch(
>             {'id': virtwho_config['id']}, output_format='base'
>         )
>         hypervisor_name, guest_name = deploy_configure_by_script(
>             script, form_data['hypervisor-type'], debug=True, org=default_org.label
>     .
>     .
> ```

@shubhamsg199 I really totally agree with you, I found there are some other redundant code in test cases, not only exist in nutanix test cases, also exist in other hypervisor, next I will submit another optimization PR, really thanks your good suggestion. I have already develop the update in local, when complete the cases automation will update the changes totally.
As the comments in https://github.com/SatelliteQE/robottelo/pull/10022 reply on https://github.com/SatelliteQE/robottelo/pull/10022#discussion_r965437106

Reply:
> > Apart from this block of code, everything is the same in the above test before this , Maybe you can add some helper functions as Griffin suggested in your UI pr or you can also consider parametrizing such tests
> > ```
> > @pytest.mark.parametrize('deploy_type', ['id', 'script'])
> > def test_positive_prism_central_deploy_configure_by_id(self, default_org, form_data, target_sat, deploy_type):
> >     .
> >     .
> >     if deploy_type == "id":
> >         command = get_configure_command(virtwho_config['id'], default_org.name)
> >         hypervisor_name, guest_name = deploy_configure_by_command(
> >             command, form_data['hypervisor-type'], debug=True, org=default_org.label
> >         )
> >     elif deploy_type == "script":
> >         script = target_sat.cli.VirtWhoConfig.fetch(
> >             {'id': virtwho_config['id']}, output_format='base'
> >         )
> >         hypervisor_name, guest_name = deploy_configure_by_script(
> >             script, form_data['hypervisor-type'], debug=True, org=default_org.label
> >     .
> >     .
> > ```
> 
> @shubhamsg199 I really totally agree with you, I found there are some other redundant code in test cases, not only exist in nutanix test cases, also exist in other hypervisor, next I will submit another optimization PR, really thanks your good suggestion. I have already develop the update in local, when complete the cases automation will update the changes totally. As the comments in #10022 reply on [#10022 (comment)](https://github.com/SatelliteQE/robottelo/pull/10022#discussion_r965437106)

@shubhamsg199 Thanks for your good suggestion, have already parametrizing the test cases using @pytest.mark.parametrize('deploy_type', ['id', 'script']), could you please help to review and merge the PR? Many thanks